TMS运单模块

fishSteak opened the pull request:
头程申请+头程单+费用明细+跟踪信息(最新)。原单类型由varchar改成ID(已完成)

fishSteak committed:
fix: 回退代码生成器修改

fishSteak commented:
头程申请+头程单+费用明细+跟踪信息(最新)。原单类型由varchar改成ID(待执行)

leefj commented on yudao-module-srm/yudao-module-srm-api/src/main/java/cn/iocoder/yudao/module/srm/config/StateMachineDescriptorRegistrar.java:
```diff
@@ -0,0 +1,11 @@
+package cn.iocoder.yudao.module.srm.config;
+
+
+public interface StateMachineDescriptorRegistrar {
```
这种方式不是太建议

leefj commented on yudao-module-srm/yudao-module-srm-biz/src/main/java/cn/iocoder/yudao/module/srm/config/BaseFailCallbackImpl.java:
```diff
@@ -6,59 +6,75 @@
-        addStateMachine(SrmPurchaseReturnDO.class, SrmAuditStatus.class, "审核退货单");
-        addStateMachine(SrmPurchaseReturnDO.class, SrmReturnStatus.class, "退款退货单");
+        // 采购申请 - 主表
+        addStateMachine(SrmPurchaseRequestDO.class, SrmOffStatus.class, "采购申请 - 开关状态");
```
这种方式不是太建议，可以调整statemachine原始代码加以支持

leefj commented on yudao-module-srm/yudao-module-srm-biz/src/main/java/cn/iocoder/yudao/module/srm/config/BaseFailCallbackImpl.java:
```diff
@@ -6,59 +6,75 @@
+
+    @Autowired(required = false)
+    public void setRegistrars(List<StateMachineDescriptorRegistrar> registrars) {
+        if (registrars == null) return;
```
建议加 {} ，不要放在一行

leefj commented on yudao-module-srm/yudao-module-srm-biz/src/main/java/cn/iocoder/yudao/module/srm/config/BaseFailCallbackImpl.java:
```diff
@@ -71,11 +87,10 @@ public void onFail(S sourceState, S targetState, E event, C context) {
 
-        log.warn("{}无法在({})状态下触发({})事件，上下文：{}", stateMachineDesc, statusDesc, SrmEventEnum.valueOf(event.toString()).getDesc(),
-            context.getClass().getName());
+        log.warn("{}无法在({})状态下触发({})事件，上下文：{}", stateMachineDesc, statusDesc, SrmEventEnum.valueOf(event.toString()).getDesc(), context.getClass().getName());
```
前办部分字符串建议使用 PURCHASE_REQUEST_NOT_EXISTS_BY_EVENT

fishSteak replied:
这是log不是业务异常

leefj commented on yudao-module-tms/yudao-module-tms-biz/src/main/java/cn/iocoder/yudao/module/tms/config/first/mile/request/TmsFirstMileRequestItemStatusMachine.java:
```diff
@@ -0,0 +1,64 @@
+@Slf4j
+@Configuration
+@SuppressWarnings({"rawtypes", "unchecked"})
+public class TmsFirstMileRequestItemStatusMachine {
```
建议以Config命名结尾

fishSteak replied:
已解决

leefj commented on yudao-module-tms/yudao-module-tms-biz/src/main/java/cn/iocoder/yudao/module/tms/config/first/mile/request/TmsFirstMileRequestStatusMachine.java:
```diff
@@ -0,0 +1,92 @@
+    }
+
+    @Resource
+    Action<TmsOffStatus, TmsEventEnum, TmsFirstMileRequestDO> requestOffActionImpl;
```
建议使用类名而非接口

fishSteak replied:
面向接口实现，可以在不修改代码的情况下后期动态替换

leefj commented on yudao-module-tms/yudao-module-tms-biz/src/main/java/cn/iocoder/yudao/module/tms/config/TmsBaseFailCallbackImpl.java:
```diff
@@ -0,0 +1,41 @@
+    public String convertEventToDescription(Object event) {
+        try{
+            if(event instanceof ArrayValuable) {
+                Method method = event.getClass().getMethod("getDesc");
```
建议使用接口，不要使用反射

fishSteak replied:
已替换

leefj commented on yudao-module-tms/yudao-module-tms-biz/src/main/java/cn/iocoder/yudao/module/tms/config/TmsStateMachineDescAutoConfig.java:
```diff
@@ -0,0 +1,32 @@
+public class TmsStateMachineDescAutoConfig implements StateMachineDescriptorRegistrar {
+
+    @Override
+    public void register(StateMachineDescriptorRegistry registry) {
```
不建议使用这种方式

leefj commented on yudao-module-tms/yudao-module-tms-biz/src/main/java/cn/iocoder/yudao/module/tms/controller/admin/first/mile/request/TmsFirstMileRequestController.java:
```diff
@@ -0,0 +1,207 @@
+        Map<Long, ErpProductDTO> productMap = erpProductApi.getProductMap(productIds);
+        Map<Long, ErpWarehouseDTO> warehouseMap = wmsWarehouseApi.getWarehouseMap(warehouseIds);
+
+        return firstMileRequestBOList.stream().map(bo -> {
```
建议简化

fishSteak replied:
装配响应VO，是这样的

leefj commented on yudao-module-tms/yudao-module-tms-biz/src/main/java/cn/iocoder/yudao/module/tms/controller/admin/first/mile/request/TmsFirstMileRequestController.java:
```diff
@@ -0,0 +1,207 @@
+        Map<Long, ErpProductDTO> productMap = erpProductApi.getProductMap(productIds);
+        Map<Long, ErpWarehouseDTO> warehouseMap = wmsWarehouseApi.getWarehouseMap(warehouseIds);
+
+        return firstMileRequestBOList.stream().map(bo -> {
```
建议简化

leefj commented on yudao-server/pom.xml:
```diff
@@ -20,14 +20,47 @@
+        </profile>
+
+        <!-- 正式环境(全部加载)-默认 -->
+        <profile>
```
不建议

leefj commented on yudao-framework/yudao-spring-boot-starter-mybatis/src/main/java/cn/iocoder/yudao/framework/mybatis/core/query/MPJLambdaWrapperX.java:
```diff
@@ -324,4 +311,41 @@ public <S, X> MPJLambdaWrapperX<T> selectLen(SFunction<S, ?> column, SFunction<X
         return this;
     }
 
+    // ========== 关键重写：使 leftJoin 返回当前类型 this ==========
```
不建议调整基础类的行为

fishSteak commented:
已解决

fishSteak committed:
refactor(somle-kingdee): 优化日志打印和供应商相关功能

- 改进日志打印格式，使用占位符方式
- 添加供应商添加功能- 为供应商名称和编码添加必填校验
-调整代码结构和注释，提高可读性

fishSteak committed:
feat(kingdee): 优化供应商数据获取效率

- 新增 getAllSupplierList 方法，使用多线程并行获取供应商数据
- 结果缓存 10 分钟，提高重复查询效率
-重构部分代码以支持新的并发逻辑

fishSteak committed:
feat(srm): 实现采购订单、入库单、退货单的同步功能

- 新增采购订单、入库单、退货单的 API接口和实现类
- 添加消息通道配置和处理逻辑
- 实现数据转换器，将 SRM 数据转换为金蝶系统所需格式
- 优化日志输出，增加进度和完成情况的记录

fishSteak committed:
feat(srm): 实现采购订单、入库单、退货单的同步功能

- 新增采购订单、入库单、退货单的 API接口和实现类
- 添加消息通道配置和处理逻辑
- 实现数据转换器，将 SRM 数据转换为金蝶系统所需格式
- 优化日志输出，增加进度和完成情况的记录

JisenCai committed:
fix(wms): 所有者批次库存-去重&库存查询-在途数量更新&日志-公司过滤项修复

JisenCai committed:
fix(wms): 代码清理+所有者批次库存日志筛选修复

fishSteak committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

# Conflicts:
#	yudao-framework/yudao-common/src/main/java/cn/iocoder/yudao/framework/common/config/ThreadPoolConfig.java

fishSteak committed:
feat(srm): 采购订单创建和更新后发送消息

- 在 SrmPurchaseOrderServiceImpl 中添加了消息发送功能
- 引入了 MessageChannel 和 MessageBuilder 用于发送消息
- 在创建采购订单和更新采购订单后，发送订单 ID 消息
- 新增了 SrmSupplierMapper 中按状态查询的功能
- 在 SrmSupplierPageReqVO 中添加了状态字段

fishSteak committed:
feat(srm): 采购到货单增加币种一致性校验- 在创建采购到货单和提交审核时，增加对关联采购订单项的币种一致性校验
- 新增错误码：采购订单项编号的币种与基准币种不一致
- 优化了代码结构，提高了采购到货单相关操作的健壮性

fishSteak committed:
refactor(srm): 优化采购订单和到货单逻辑

- 修改采购申请项错误提示，使用明细行编号
- 优化采购到货单审核流程，处理关联入库单
- 按仓库分组生成入库单
-增加采购申请单号字段
- 优化采购订单项创建逻辑，校验申请项存在性

fishSteak committed:
feat(srm): 优化采购入库单创建逻辑

- 在采购入库单中添加采购订单编号字段- 为采购入库明细项添加采购申请人和申请部门字段
- 优化采购入库单创建流程，支持批量创建入库明细项
- 更新 WmsInboundItemSaveReqDTO 和 WmsInboundSaveReqDTO 类结构

fishSteak committed:
refactor(srm): 重构采购入库逻辑

- 修改 inItemActionImpl 类，增加与订单项的状态机交互
- 修正 SrmPurchaseInCountDTO 类名
- 更新 SrmPurchaseInItemStatusMachine 类中的状态机配置
-优化 SrmSupplierProductServiceImpl 类的导入顺序和格式
-调整 SrmAuditStatus 枚举类的注释

fishSteak committed:
refactor(srm): 优化采购入库状态机逻辑

- 更新 InItemActionImpl 类，增加到货单明细行入库状态机注释
- 重构 ItemStorageActionImpl 中的 execute 方法，明确更新入库数量和状态的逻辑
-调整 OrderItemInActionImpl 中的逻辑，直接使用 dtoCount 而非计算差额- 修改 SrmPurchaseInCountDTO 中的字段注释，提高代码可读性
- 增加 SrmPurchaseInItemMapper 中的查询方法，提高数据访问效率

JisenCai committed:
fix(wms): 在途数更新 + 生成入库流水单 + 代码清理

JisenCai committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

fishSteak committed:
feat(srm): 优化采购到货单入库逻辑

- 新增总毛重和总体积字段- 完善采购订单项和主表信息获取
- 设置库存财务公司ID
- 优化入库单创建逻辑
- 更新采购订单保存请求VO

JisenCai committed:
fix(tms):公司ID校验

fishSteak committed:
feat(srm): 优化采购到货单入库逻辑

- 新增总毛重和总体积字段- 完善采购订单项和主表信息获取
- 设置库存财务公司ID
- 优化入库单创建逻辑
- 更新采购订单保存请求VO

fishSteak committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

fishSteak committed:
feat(srm): 优化采购到货单入库逻辑

- 新增总毛重和总体积字段- 完善采购订单项和主表信息获取
- 设置库存财务公司ID
- 优化入库单创建逻辑
- 更新采购订单保存请求VO

fishSteak committed:
refactor(srm): 优化采购订单和请求相关功能

- 更新采购订单转换逻辑，增加产品含税单价字段
- 调整采购订单保存请求 VO，移除产品单位 ID 必填验证
- 优化采购请求合并逻辑，限制异常信息长度
-移除采购请求条目保存请求 VO 中的税额必填验证

fishSteak committed:
refactor(srm): 优化采购入库和订单服务中的价格计算和状态校验

- 在 SrmPurchaseInServiceImpl 中添加安全处理 BigDecimal 的方法，避免空指针异常
- 优化 SrmPurchaseOrderServiceImpl 中的采购订单状态校验逻辑

fishSteak committed:
refactor(srm): 优化采购入库和订单服务中的价格计算和状态校验

- 在 SrmPurchaseInServiceImpl 中添加安全处理 BigDecimal 的方法，避免空指针异常
- 优化 SrmPurchaseOrderServiceImpl 中的采购订单状态校验逻辑

fishSteak committed:
refactor(srm): 优化采购入库和订单服务中的价格计算和状态校验

- 在 SrmPurchaseInServiceImpl 中添加安全处理 BigDecimal 的方法，避免空指针异常
- 优化 SrmPurchaseOrderServiceImpl 中的采购订单状态校验逻辑

fishSteak committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

fishSteak committed:
feat(srm): 采购订单执行状态动态调整

- 新增执行调整事件 EXECUTION_ADJUSTMENT- 实现执行调整逻辑，允许在不同执行状态下进行动态调整- 更新状态机配置，支持执行调整事件

fishSteak committed:
feat(srm): 采购订单执行状态动态调整

- 新增执行调整事件 EXECUTION_ADJUSTMENT- 实现执行调整逻辑，允许在不同执行状态下进行动态调整- 更新状态机配置，支持执行调整事件

fishSteak committed:
fix(srm): 修正采购入库单关联错误

- 将 WmsInboundApi.getInboundList 方法的参数从 BillType.WMS_INBOUND 修改为 BillType.SRM_PURCHASE_IN
- 此修改确保了正确的单据类型被用于查询关联的入库单模块的交互，使用更精确的枚举值

JisenCai committed:
feat(tms): srm向wms创建/废除入库单据

fishSteak committed:
feat(wms): 采购到货单反审核时作废已创建的入库单，并调整库存数量- 在 yudao-module-wms-biz 中添加 yudao-module-srm-api 依赖
- 修改 SrmPurchaseInServiceImpl 中的反审核逻辑，作废相关入库单
- 更新 WmsInboundApi 接口，添加 billType 参数
- 实现 WmsInboundApiImpl 中的作废入库单和调整库存逻辑

fishSteak committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

fishSteak committed:
refactor(srm): 优化采购到货单审核通过后的入库单创建逻辑

- 在创建入库单后，增加日志记录入库单 ID
- 修改 abandonInbound 方法，增加 billType 参数并处理相关逻辑
- 优化到货单明细行的入库状态机事件触发逻辑

fishSteak committed:
refactor(srm): 将采购入库单的日志信息改为采购到货单

- 修改了 SrmPurchaseInServiceImpl 类中创建、更新、删除和审核采购入库单的日志信息
- 将"采购入库单"改为"采购到货单"，以更准确地反映业务流程
- 优化了 WmsInboundApiImpl 类中的代码格式

fishSteak committed:
feat(srm): 优化采购入库单状态管理

- 新增入库状态字段和相关枚举
- 实现入库单和入库项的状态同步
- 优化状态机设计，支持部分入库和未入库状态
- 调整接口和消息结构，提高系统间协同效率

fishSteak committed:
refactor(srm): 重构采购到货单相关代码

- 新增 SrmPurchaseInBO 和 SrmPurchaseInItemBO 类用于封装采购到货单和到货项数据
- 修改 SrmPurchaseInController 中的接口返回类型为使用新的 BO 类
- 重构 SrmPurchaseInService 接口，增加新的查询方法
- 实现 SrmPurchaseInServiceImpl 中的 BO 查询逻辑
- 更新 SrmPurchaseInItemMapper 中的查询方法，支持关联查询

JisenCai committed:
feat(wms): 根据入库单生成出库单

fishSteak committed:
feat(srm): 优化采购退货功能

- 移除了不必要的注释代码
- 调整了字段顺序和命名，使其更加合理
- 增加了供应商信息、审核信息等新字段
- 优化了退货项的创建和更新逻辑
- 新增了总体积和总重量的计算
- 改进了错误处理和数据校验

fishSteak committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

fishSteak committed:
feat(srm): 优化采购退货功能并更新 WMS 出库相关 API

- 在采购退货请求中添加至少选择一个到货明细行的校验
- 更新 WMS 出库 API，使用新的 SrmPurchaseInItemCountDTO 替代 SrmPurchaseInCountDTO

fishSteak committed:
refactor(srm): 优化采购到货接口和数据模型

- 将 SrmPurchaseInController 中的 GET 请求改为 POST 请求，以适应分页查询
- 更新请求参数注解以适应新的请求方式
- 修改 SrmPurchaseReturnSaveReqVO 中的字段描述，提高代码可读性

fishSteak committed:
feat(srm): 采购到货新增关联订单数量字段并优化相关功能

- 在 SrmPurchaseInBaseRespVO 中添加 orderQty 字段，用于显示关联订单行的采购数
- 更新 SrmPurchaseInController 中的查询接口，增加关联订单数量的查询逻辑
- 修改 SrmPurchaseInItemMapper 中的查询条件，支持按审核状态筛选
- 更新 SrmPurchaseInPageReqVO，增加 auditStatus 字段用于筛选审核状态
-调整 SrmPurchaseInSaveReqVO 中 id 字段的校验逻辑，确保创建和更新时的正确性
- 优化 SrmPurchaseOrderSaveReqVO 中 id 字段的校验逻辑，确保创建和更新时的正确性- 更新 SrmPurchaseReturnController 中的创建和更新接口，增加 Validation.OnCreate 和 Validation.OnUpdate 分组的校验逻辑
-调整 SrmPurchaseReturnSaveReqVO 中 id 字段的校验逻辑，确保创建和更新时的正确性

JisenCai committed:
fix(wms):到货单状态机触发变更 & 出库单作废

fishSteak committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

fishSteak committed:
feat(purchase): 优化采购退货功能

- 新增退货项对应的入库单供应商一致性校验
- 完善退货单和退货项的相关逻辑
- 优化采购到货分页查询接口
- 修改采购订单分页查询为 POST 请求

fishSteak committed:
feat(srm): 退货单审核时创建 WMS 出库单

- 在退货单审核通过时，创建 WMS 出库单
- 实现了从退货单到出库单的数据转换和接口调用
- 目前代码已注释，需要进一步开发和测试

fishSteak committed:
feat(srm): 退货单审核时创建 WMS 出库单

- 在退货单审核通过时，创建 WMS 出库单
- 实现了从退货单到出库单的数据转换和接口调用
- 目前代码已注释，需要进一步开发和测试

fishSteak committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

# Conflicts:
#	yudao-module-wms/yudao-module-wms-api/src/main/java/cn/iocoder/yudao/module/wms/api/outbound/dto/WmsOutboundSaveReqDTO.java

fishSteak committed:
feat(srm): 优化采购订单、入库单相关功能

-为 SrmPurchaseOrderDO 添加币别 ID 字段
- 修复 SrmPurchaseOrderSaveReqVO 中的币别相关字段
-优化采购订单、入库单、请购单等的排序逻辑
- 调整 TmsFirstMileConvert 中的出库单转换逻辑

JisenCai committed:
feat(wms): 列表查询货物以及库存详情

fishSteak committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

fishSteak committed:
feat(srm): 采购到货单关联采购订单的供应商和采购公司校验

- 新增校验逻辑，确保采购到货单关联的采购订单具有相同的供应商和采购公司
- 添加对应的错误码和异常处理
- 优化了代码结构，提高了可读性和可维护性

JisenCai committed:
fix(wms): 添加DTO备注

JisenCai committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

JisenCai committed:
fix(wms): 添加DTO备注

fishSteak committed:
feat(srm): 采购到货单关联采购订单的供应商和采购公司校验

- 新增校验逻辑，确保采购到货单关联的采购订单具有相同的供应商和采购公司
- 添加对应的错误码和异常处理
- 优化了代码结构，提高了可读性和可维护性

fishSteak committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

JisenCai committed:
fix(wms): 退货单触发出库单时的库位选择逻辑调整 & 代码清理

JisenCai committed:
fix(wms): 更新WmsOutboundImportReqDTO

fishSteak committed:
feat(srm): 采购退货单反审核功能

- 新增采购退货单反审核逻辑
-增加 WMS 出库单生成和作废功能
- 优化采购退货单审核流程- 调整 WMS出库单相关 DTO 结构

fishSteak committed:
fix(srm): 优化采购退货审核出库单创建逻辑

- 新增错误码 PURCHASE_RETURN_PROCESS_FAIL_WMS_OUTBOUND_EXISTS，用于处理创建出库单失败的情况- 在审核退货单时，添加创建出库单的异常捕获，提高系统稳定性
- 在 WmsOutboundItemSaveReqDTO 中重新启用 companyId 和 deptId 字段
- 在 WmsOutboundType 中添加 OUTBOUND_BILL 类型，用于头程单出库

JisenCai committed:
feat(wms):出库动作状态机变更 + 出库单生成

fishSteak committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

fishSteak committed:
fix(srm): 修复采购退货单生成出库单异常

- 移除了不必要的代码和导入- 优化了 createWmsOutbound 方法，按仓库分组退货项并为每个仓库创建出库单
- 添加了对仓库 ID 为空的异常处理

fishSteak committed:
refactor(srm): 优化采购退货单反审核逻辑

- 提取作废 WMS 出库单逻辑到单独的方法 abandonWmsOutbound 中
- 在采购退货单反审核和撤回审核时统一调用 abandonWmsOutbound 方法
- 优化代码结构，提高可维护性和复用性

fishSteak committed:
refactor(srm): 优化采购订单相关功能

- 移除 SrmPurchaseOrderItemMapper 中的 orderByDesc 调用
- 在 SrmPurchaseOrderMapper 中添加对 containerRate 和 purchaseApplyCode 的模糊查询
- 删除 SrmPurchaseOrderPageReqVO 中的 fileUrl 字段
- 将 SrmPurchaseRequestController 中的 GET 请求改为 POST 请求
- 优化用户信息获取逻辑

fishSteak committed:
refactor(srm): 重构采购退货单查询接口

- 新增 SrmPurchaseReturnBO 和 SrmPurchaseReturnItemBO 类用于封装退货单和退货项信息
- 修改 SrmPurchaseReturnController 中的查询接口，使用新的 BO 类作为返回值
- 在 SrmPurchaseReturnService 接口中新增 getPurchaseBOReturn 和 getPurchaseReturnBOPage 方法
- 实现 SrmPurchaseReturnServiceImpl 中的 getPurchaseBOReturn 和 getPurchaseReturnBOPage 方法
- 新增 SrmPurchaseReturnConvert 类用于转换退货单和退货项数据rmPurchaseReturnPageReqVO 中添加创建人字段

fishSteak committed:
refactor(srm): 优化采购退货单查询性能

- 在 SrmPurchaseReturnController 中添加 returnItemMap，提高退货项查询效率- 在 SrmPurchaseReturnItemMapper 中增加对空查询参数的处理，提升代码健壮性

fishSteak committed:
feat(srm): 新增采购退货出库功能

- 新增出库状态枚举和相关DTO- 实现出库状态机和出库操作逻辑
- 重构退款相关代码，改为退货处理
- 优化数据库表结构，增加出库相关字段

JisenCai committed:
fix(wms): 更新分页接口为POST方法

JisenCai committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

JisenCai committed:
fix(wms): 创建时间允许范围查询

fishSteak committed:
feat(purchase): 退货单新增出库状态字段

- 在 SrmPurchaseReturnBaseRespVO 中添加出库状态字段 outboundStatus
- 在 SrmPurchaseReturnItemMapper 中增加对出库状态的查询条件
- 在 SrmPurchaseReturnPageReqVO 中添加出库状态查询参数 outboundStatus

JisenCai committed:
fix(wms): 创建时间筛选 & 退货单状态机更新

fishSteak committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

fishSteak committed:
refactor(srm): 重构采购订单相关代码

- 优化了 SrmPurchaseInServiceImpl 中的 getPurchaseInBOById 方法
-调整了 SrmPurchaseOrderBaseRespVO 的结构，使其更加合理
- 修改了 SrmPurchaseOrderItemMapper 中的查询条件- 更新了 SrmPurchaseOrderMergeReqVO 的结构，使其更符合业务逻辑
- 调整了 SrmPurchaseOrderServiceImpl 中的 merge 方法

JisenCai committed:
fix(wms): 时间格式调整（以后可能会使用筛选）

JisenCai committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

fishSteak committed:
refactor(tms): 优化头程单创建逻辑

- 移除控制器中的 initId 方法调用
- 在服务实现类中处理上游 ID 的设置
- 优化船期信息保存逻辑，增加异常处理
- 调整费用项保存方法，提高代码可读性
- 新增头程单创建失败的错误码

JisenCai committed:
fix(tms): 船运数据未能带出问题修复

fishSteak committed:
refactor(tms): 重构头程管理查询逻辑

- 优化了 FistMileApiImpl 中的 getFirstMile 方法，使用 BeanUtils.toBean 替代转换类
- 修正了 TmsFeeMapper 中的 selectByIdAndType 方法，返回类型由单个对象改为列表
- 更新了 TmsFeeService 接口中的 getFee 方法，返回类型改为列表
-重构了 TmsFeeServiceImpl 中的 getFee 方法，适应返回列表的变更
- 删除了 TmsFirstMileConvert 类中未使用的转换方法
- 优化了 TmsFirstMileItemMapper 中的查询逻辑，使用新构建的 wrapper- 移除了 TmsFirstMileMapper 中的冗余方法- 更新了 TmsFirstMilePageReqVO 类的结构，将主查询条件提取到单独的类中
- 调整了 TmsFirstMileRequestServiceImpl 中的依赖注入方式

fishSteak committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

JisenCai committed:
fix(tms): 代码清理

fishSteak committed:
refactor(srm): 重构采购订单状态机逻辑

- 优化了采购订单状态机的配置，简化了状态转换逻辑
- 添加了币别 ID 字段和相关逻辑处理
- 调整了入库时间和币别的设置方式- 移除了不必要的注释和未使用的代码

fishSteak committed:
refactor(tms): 优化头程单反审核逻辑

- 添加作废 WMS 出库单的方法
- 完善头程单反审核时的出库单处理逻辑
- 更新错误码枚举类，增加新的错误码- 优化代码结构和注释

fishSteak committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

JisenCai committed:
fix(wms): 入库单类型，查询，分页逻辑调整

fishSteak committed:
refactor(tms): 重构头程单费用处理逻辑

- 新增 TmsFeeMapper 中的 deleteBatchIdsBySourceType 和 deleteBySourceIdAndSourceType 方法
- 修改 TmsFeeService 接口，移除冗余方法，增加新的删除方法
- 重构 TmsFeeServiceImpl 中的费用处理逻辑，使用新的删除方法
- 优化 TmsFirstMileConvert 中的跟踪信息处理
- 增加 TmsFirstMileService 接口中的 validateFirstMileExists 方法
- 重构 TmsFirstMileServiceImpl 中的头程单处理逻辑，使用新的费用处理方法

fishSteak committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

# Conflicts:
#	yudao-module-tms/yudao-module-tms-biz/src/main/java/cn/iocoder/yudao/module/tms/convert/first/mile/TmsFirstMileConvert.java

fishSteak committed:
refactor(tms): 重构头程管理转换逻辑

- 移除了 TmsFirstMileRequestConvert 类中的冗余方法
-优化了 TmsFirstMileRequestServiceImpl 和 TmsFirstMileServiceImpl 中的对象转换逻辑
- 使用 BeanUtils.toBean() 替代自定义转换方法，简化代码并提高可维护性

JisenCai committed:
fix(wms): 出库单人员名单补充

JisenCai committed:
fix(wms): 增加计划出库单时间字段

fishSteak committed:
refactor(tms): 重构头程单和费用处理逻辑

- 移除头程单中的提单号、预计送仓时间和实际送仓时间字段
- 优化费用处理逻辑，使用泛型处理不同的费用请求
- 调整船运跟踪信息的处理方式
- 重构代码结构，提高可维护性和可扩展性

JisenCai committed:
fix(wms): 出库单创建及审批时加入出库时间

fishSteak committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

fishSteak committed:
refactor(tms): 移除 TmsTransferMapper 中的冗余方法

- 删除了 TmsTransferMapper 接口中的 selectPage 方法
- 该方法可能已在其他地方实现了相同功能，或者不再需要

fishSteak committed:
feat(tms): 创建调拨单功能

- 新增调拨单创建接口和相关服务
- 实现调拨单明细的批量插入
- 添加调拨单总数量、总重量等信息的计算逻辑
- 优化调拨单更新逻辑，支持明细的增删改
- 新增调拨单删除接口，同时删除相关明细

fishSteak committed:
refactor(tms): 重构调拨单相关代码

- 新增 TmsTransferBO 和 TmsTransferItemBO 类作为业务对象
- 新增 TmsTransferRespVO 和相关 Common VO 类作为响应对象- 重构 TmsTransferService接口，新增 BO 和 VO相关方法
- 实现 TmsTransferServiceImpl 中的 BO 和 VO 相关方法
- 新增 TmsTransferConvert 类用于转换 BO 和 DO
- 优化 TmsTransferItemMapper 接口，新增关联查询方法
- 更新 TmsTransferController 中的接口返回类型为 VO

JisenCai committed:
外部接口-更新在制数量

JisenCai committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

fishSteak committed:
refactor(srm): 优化采购入库单和订单状态机配置- 简化状态机配置代码，使用静态导入
-优化状态机事件处理逻辑，提高可读性
- 调整部分状态机事件处理方式，提高灵活性

fishSteak committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

fishSteak committed:
refactor(srm): 重构采购模块状态机

- 更新状态机常量引用，使用新的枚举值
- 优化状态机注册方式，使用 @Resource 注解- 统一状态机事件处理逻辑
- 调整部分类名和变量名，提高可读性

fishSteak committed:
refactor(srm): 重构采购模块状态机相关代码

-优化了多个类中的状态机配置和使用
- 新增 SrmOrderItemOffDTO 类用于订单项开关状态
- 调整了状态机 Action 的注入方式- 修复了一些状态机相关的 bug

fishSteak committed:
refactor(srm): 重构采购模块状态机相关代码

-优化了多个类中的状态机配置和使用
- 新增 SrmOrderItemOffDTO 类用于订单项开关状态
- 调整了状态机 Action 的注入方式- 修复了一些状态机相关的 bug

fishSteak committed:
refactor(srm): 重构采购入库相关代码

- 将 SrmRequestInDTO 重命名为 SrmRequestInMachineDTO
-优化了采购入库状态机的实现
- 重构了采购入库审核逻辑，拆分为审核和反审核两个独立的处理流程
-优化了与仓库管理系统(WMS)的交互逻辑

JisenCai committed:
fix(wms): 反审核逻辑更新 & srm触发入库单逻辑更新

JisenCai committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

# Conflicts:
#	yudao-module-srm/yudao-module-srm-biz/src/main/java/cn/iocoder/yudao/module/srm/service/purchase/impl/SrmPurchaseInServiceImpl.java

JisenCai committed:
fix(wms): 反审核逻辑更新 & srm触发入库单逻辑更新

JisenCai committed:
fix(wms): 反审核逻辑更新 & srm触发入库单逻辑更新

JisenCai committed:
fix(wms): 增加InboundDTO的明细行

fishSteak committed:
feat(srm): 实现采购入库单审核回调接口

- 新增 SrmInboundReqDTO 和 SrmInboundItemReqDTO 类用于处理入库单审核回调
- 实现 updatePurchaseInItemQty 方法，处理入库单审核后的数量变动- 添加状态机相关逻辑，用于更新入库项状态
- 优化代码结构，提高可维护性和可扩展性

fishSteak committed:
feat(tms): 实现调拨单审核功能

- 新增调拨单审核状态机和相关动作类
- 实现调拨单提交审核、审核通过、审核不通过和反审核等功能
-优化调拨单审核流程，支持状态机管理审核状态

fishSteak committed:
feat(srm): 接口调整支持 WMS 出库单审核回调

- 新增 SrmOutboundReqDTO 和 SrmOutboundItemReqDTO 类用于 WMS出库单回调
- 修改 SrmPurchaseReturnApi 接口，增加对 WMS 出库单审核回调的支持
- 实现 SrmPurchaseReturnApiImpl 中的 WMS 出库单审核回调逻辑
- 在 SrmPurchaseReturnService 中添加校验采购退货项是否存在的方法
- 在 SrmPurchaseReturnServiceImpl 中实现校验采购退货项是否存在的方法

fishSteak committed:
fix(srm): 修复采购入库事件参数传递问题

- 将 SrmPurchaseInItemCountDTO 的构建过程优化，使用变量存储构建结果
- 在状态机事件触发时，正确传递 SrmPurchaseInItemCountDTO 对象作为参数

JisenCai committed:
fix(wms): 增加InboundDTO的明细行

fishSteak committed:
feat(tms): 实现调拨出库功能

- 新增 TmsTransferApi 接口，定义出库单审核后回调方法- 实现 TmsTransferApiImpl 类，处理出库单审核回调逻辑
- 在 TmsTransferServiceImpl 中添加创建 WMS 出库单和作废出库单的方法
- 新增 TmsOutboundReqDTO 和 TmsOutboundItemReqDTO 类，用于处理出库单相关数据
- 在 WmsOutboundType 枚举中添加 TRANSFER 类型，表示调拨出库

fishSteak committed:
feat(tms): 调拨单出库审核后创建入库单

- 在 TmsTransferApiImpl 类中添加创建入库单的逻辑
- 新增与 WmsInboundApi 的交互，调用 createInbound 方法创建入库单
- 将调拨单信息转换为入库单信息，包括主单和明细项
- 记录创建入库单的日志信息

fishSteak committed:
refactor(srm): 调整采购订单状态机和存储状态枚举

- 修改采购订单状态机中的状态转换逻辑，简化了库存调整相关的外部转换
- 在 SrmStorageStatus 枚举类中添加 getAllStatuses 方法，方便获取所有存储状态

fishSteak committed:
refactor(tms): 重构调拨单状态更新逻辑

- 新增 TmsTransferStatusUpdateDTO 类用于更新调拨单状态
- 在 TmsTransferService 接口中添加 updateTransferStatus 方法
- 在 TmsTransferServiceImpl 中实现 updateTransferStatus 方法
- 更新 TmsTransferApi 和 TmsTransferApiImpl 中的出库单审核后回调逻辑
- 新增调拨单明细出库信息更新功能
-优化反审核逻辑

fishSteak committed:
feat(tms): 实现调拨入库回调功能

- 新增 TmsInboundReqDTO 和 TmsInboundItemReqDTO 类用于处理调拨入库请求
- 在 TmsTransferApi 中添加 afterInboundAudit 方法处理入库回调
- 在 TmsTransferServiceImpl 和 TmsTransferItemServiceImpl 中实现相关业务逻辑- 优化了数据校验和事务处理

fishSteak committed:
refactor(srm): 优化采购模块相关 Mapper 的排序逻辑

- 修改 SrmPurchaseInItemMapper、SrmPurchaseOrderItemMapper、SrmPurchaseRequestItemsMapper 和 SrmPurchaseReturnItemMapper
- 移除对未使用的 SrmPurchaseOrderDO 的导入
- 将 orderByDesc 方法的参数从 SrmPurchaseOrderDO 改为对应的实体类字段
- 在 SrmPurchaseOrderItemMapper 和 SrmPurchaseRequestItemsMapper 中添加按创建时间降序排序

fishSteak committed:
refactor(srm): 优化采购订单编号生成和校验逻辑- 修改采购订单编号校验规则，支持手动输入编号
-优化采购订单编号生成逻辑，避免重复生成
- 更新错误提示信息，使其更加准确
- 移除不必要的正则表达式校验

fishSteak committed:
fix(srm): 优化采购到货单删除逻辑

- 修改删除失败的错误提示，明确只有草稿、未通过、审核撤销状态的到货单可以删除
- 优化删除逻辑，提高代码可读性和性能
- 移除不必要的注释和空行

c-tao commented on somle-esb/src/main/java/com/somle/esb/config/IntegrationConfig.java:
```diff
@@ -6,15 +6,16 @@
 
 @Slf4j
 @Configuration
+@EnableJpaRepositories(basePackages = "com.somle")  //esb模块才加载com.somle
```
撤销这个，未来包名和orm都会统一

c-tao commented on somle-esb/src/main/java/com/somle/esb/convert/KingdeePurInboundConvert.java:
```diff
@@ -0,0 +1,34 @@
+ * 采购入库单转换器
+ */
+@Mapper
+public interface KingdeePurInboundConvert {
```
跨系统的转换器不使用mapstruct，放到ErpToKingdeeConverter去

c-tao commented on somle-esb/src/main/java/com/somle/esb/job/KingdeePurInboundDataJob.java:
```diff
@@ -13,7 +13,7 @@ public String execute(String param) throws Exception {
         var vo = KingdeePurInboundReqVO.builder().build();
         for (KingdeeClient client : kingdeeService.getClients()) {
-            client.streamPurInbound(vo).forEach(
+            client.getAllPurInbound(vo).forEach(
```
参考阿里规范，各层命名规约

c-tao commented on somle-kingdee/src/main/java/com/somle/kingdee/constant/ErpRedisKeyConstants.java:
```diff
@@ -0,0 +1,20 @@
+package com.somle.kingdee.constant;
+
+/**
+ * ERP系统 Redis Key 常量类
```
拷过来没有改？

c-tao commented on somle-kingdee/src/main/java/com/somle/kingdee/constant/ErpRedisKeyConstants.java:
```diff
@@ -0,0 +1,20 @@
+     * 金蝶物料缓存
+     * 格式：kingdee:material:{物料编号}
+     */
+    public static final String KINGDEE_MATERIAL = "esb:kingdee:material";
```
kingdee不是esb的子模块1

c-tao commented on somle-kingdee/src/main/java/com/somle/kingdee/model/supplier/KingdeeSupplier.java:
```diff
@@ -2,18 +2,13 @@
+
 @Data
 @JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
 public class KingdeeSupplier {
```
类的VO后缀

c-tao replied:
RespVO?

fishSteak replied:
这个使用的地方太多了，后面再看一看

c-tao commented on somle-kingdee/src/main/java/com/somle/kingdee/model/vo/KingdeeSupplierQueryReqVO.java:
```diff
@@ -0,0 +1,105 @@
+ */
+@Data
+@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
+public class KingdeeSupplierQueryReqVO {
```
KingdeeSupplierReqVO？

fishSteak replied:
除了 query还有 saveVO呢

JisenCai committed:
fix(wms): 修复建入库单报错&更新wms单据号生成规则&代码清理

JisenCai committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

fishSteak committed:
chore: 删除 yudao-server-test 项目的配置文件

删除了 yudao-server-test 项目中的三个配置文件：
- application.yaml- application-dev.yaml
- application-local.yaml

这些文件包含了数据库、Redis、消息队列、微信公众号等大量配置信息，删除它们可能会对项目功能产生重大影响。请确保在其他地方备份了这些配置信息。

fishSteak committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

JisenCai committed:
feat(wms): 调拨单生成的出库/入库单进行回调动作

fishSteak committed:
feat(tms): 添加调拨单编码唯一性校验

- 新增调拨单编码重复错误码和异常处理
- 在创建和更新调拨单时增加编码唯一性校验
- 优化 TmsProductRespVO 中字段命名- 更新 TmsTransferSaveReqVO 中的注解
- 移除 pom.xml 中的冗余配置

fishSteak committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

JisenCai committed:
feat(wms): 查询库存可售数量接口

JisenCai committed:
feat(wms): 查询库存可售数量接口

fishSteak committed:
feat(tms): 添加获取可售库存数量功能

- 新增 TmsTransferSellableQtyReqVO 和 TmsTransferSellableQtyRespVO 类用于可售库存查询
- 在 TmsTransferController 中添加 getSellableQty 接口
- 在 TmsTransferService 中添加 getSellableQty 方法
- 在 TmsTransferServiceImpl 中实现 getSellableQty 方法，调用 WMS接口获取可售库存
- 优化单据编号格式验证规则
- 移除不必要的注解和导入

fishSteak committed:
refactor(tms): 优化调拨审核状态机和相关功能- 更新 TmsTransferImpl 中的状态机调用逻辑
- 为 TmsTransferSaveReqVO 中的 code 字段添加 @NotNull 注解
- 修正 TmsTransferStatusMachine 中的状态机配置

fishSteak committed:
refactor(tms): 优化调拨审核状态机和相关功能- 更新 TmsTransferImpl 中的状态机调用逻辑
- 为 TmsTransferSaveReqVO 中的 code 字段添加 @NotNull 注解
- 修正 TmsTransferStatusMachine 中的状态机配置

c-tao commented on yudao-framework/yudao-common/src/main/java/cn/iocoder/yudao/framework/common/config/ThreadPoolConfig.java:
```diff
@@ -0,0 +1,39 @@
+ * 线程池配置类
+ */
+@Configuration
+public class ThreadPoolConfig {
```
删除这个类，用到的地方显式从AsynTask拿

c-tao replied:
StatusValue不做公共结构

c-tao commented on yudao-framework/yudao-common/src/main/java/cn/iocoder/yudao/framework/common/enums/enums/DictTypeConstants.java:
```diff
@@ -11,5 +11,21 @@ public interface DictTypeConstants {
     //国家代码
     String COUNTRY_CODE = "country_code";
 
+    //开启状态，布尔值
```
这些全部删除·

c-tao commented on yudao-framework/yudao-common/src/main/java/cn/iocoder/yudao/framework/common/enums/ChannelEnum.java:
```diff
@@ -0,0 +1,28 @@
+ * 消息通道枚举
+ */
+public interface ChannelEnum {
+
```
全部删除

JisenCai committed:
fix(wms): 更新盘点单代码移除&出库单逻辑调整

fishSteak committed:
refactor(srm): 重构 SRM 模块代码

-移除 AbsoluteDeleteMpSqlInjector 类
- 更新 BaseFailCallbackImpl 中的事件描述获取方式
- 删除 BaseMapperX 中的物理删除方法
- 移除 ChannelEnum接口
- 更新 CRM、FMS、SRM 模块中的字典类型常量
-重构 KingdeeClient 中的线程池使用
- 移除 RestTemplateConfig 类
- 更新 SRM 模块中的枚举类结构
- 调整 SRM 消息通道配置

JisenCai committed:
fix(wms): 与tms交互报错

JisenCai committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

fishSteak committed:
refactor(framework): 优化 MyBatis、定时任务和消息队列相关配置

- 移除了 MyBatis 中的 AbsoluteDeleteMpSqlInjector 相关代码
- 在 Quartz 配置中添加了 EnableScheduling 注解
-调整了 Redis MQ 消费配置，移除了不必要的条件判断

fishSteak committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

fishSteak committed:
Merge branch 'somle-test' into somle-test-dev-TMS

# Conflicts:
#	yudao-module-srm/yudao-module-srm-biz/src/test/java/cn/iocoder/yudao/module/erp/dal/mysql/purchase/SrmPurchaseInMapperTest.java
#	yudao-module-system/yudao-module-system-api/src/main/java/cn/iocoder/yudao/module/system/api/user/AdminUserApi.java

fishSteak committed:
refactor(framework):  类型

fishSteak committed:
refactor(framework): 优化 MyBatis、定时任务和消息队列相关配置

- 移除了 MyBatis 中的 AbsoluteDeleteMpSqlInjector 相关代码
- 在 Quartz 配置中添加了 EnableScheduling 注解
-调整了 Redis MQ 消费配置，移除了不必要的条件判断

fishSteak committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

fishSteak committed:
refactor(system): 优化 AdminUserApiImpl 类并添加 getUsernameIndex 方法

- 移除了未使用的 import 语句
- 添加了 getUsernameIndex 方法，目前返回 null
- 优化了 createUser 方法，直接使用 userService 创建用户
- 修改了 getOrderUnarchivePages 方法的签名，添加了多余的空格

fishSteak committed:
feat(tms): 新增库存归属部门字段并优化相关功能

- 新增 TmsDeptRespVO 类用于表示部门信息
- 在 TmsTransferItemDO 和相关 VO 中添加 deptId 字段
- 实现部门信息的装配和展示- 优化状态机异常提示信息
- 调整日志输出格式

fishSteak committed:
refactor(tms): 重构组装数据的方法

- 新增 TmsStreamXTool 类，实现通用的组装方法
- 重构 assembleProducts、assembleCompany、assembleWarehouse 和 assembleDept 方法，使用新的组装工具- 优化代码结构，提高可维护性和可重用性

JisenCai committed:
fix(wms): 调拨单生成出库单添加缺省公司id

fishSteak committed:
refactor(wms): 重构 WMS 模块代码

- 更新了 InboundAgreeTransitionHandler 和 OutboundAgreeTransitionHandler 中的请求 DTO 类型
- 重命名了 SrmInboundReqDTO 和 SrmOutboundReqDTO 类
- 调整了 SrmPurchaseInApi 和 SrmPurchaseReturnApi 接口中的方法参数
- 移除了不必要的字段和方法，简化了数据结构

fishSteak committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

JisenCai committed:
fix(wms): 开放可售数量查询接口

JisenCai committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

fishSteak committed:
refactor(srm): 重构 SRM 模块的上下文对象

- 将多个 SrmXXXDTO 类重命名为 SrmXXXContext 类
- 更新相关的 Action 和 StateMachine 接口引用
-调整包结构，将 API 相关的类移动到 config 包下

fishSteak committed:
feat(tms): 添加港口名称重复校验

- 在创建和更新港口信息时增加名称重复校验逻辑
- 新增 PORT_INFO_NAME_DUPLICATE 错误码用于处理港口名称重复的情况
- 优化代码结构，提高可维护性

JisenCai committed:
fix(wms): 查询数量接口返回结构调整

JisenCai committed:
fix(wms): 查询数量接口返回结构调整

fishSteak committed:
refactor(wms): 移除 WmsInboundServiceImpl 中未使用的 SRM 相关代码

- 删除了未使用的 SRM 相关导入
- 移除了未使用的 purchaseInCountDTOStateMachine 字段

c-tao commented on yudao-module-tms/yudao-module-tms-biz/src/main/java/cn/iocoder/yudao/module/tms/controller/admin/first/mile/request/TmsFirstMileRequestController.java:
```diff
@@ -0,0 +1,280 @@
+    @Operation(summary = "获取产品可用库存")
+    @PreAuthorize("@ss.hasPermission('tms:first-mile-request:query')")
+    public CommonResult<TmsFirstMileRequestProductStockRespVO> getProductStock(@Valid @RequestBody TmsFirstMileRequestProductStockReqVO reqVO) {
+        Map<Long, WmsStockOwnershipDTO> stockMap = wmsStockOwnershipApi.selectByDeptIdAndProductIdAndCountryIdMap(reqVO.getDeptId(), reqVO.getProductIds(), reqVO.getCountry());
```
这个方法太长了！

fishSteak replied:
getProductStock 只有10行   有个绑定渲染多了一点

fishSteak committed:
refactor(tms): 重构头程单相关代码

- 修改类名和包名，统一命名规范
- 优化部分代码结构，提高可读性
- 移除未使用的 LogRecordConstants 接口

c-tao committed:
fix: 将EnableScheduling注解移到独立的autoConfiguration

fishSteak committed:
refactor(converter): 重构数据转换逻辑

- 移除原有的 KingdeePurInboundConvert、KingdeePurOrderConvert、KingdeePurReturnConvert、KingdeeSupplierConvert 接口
- 新增 ErpToKingdeeConverter 类，统一处理所有类型的转换
- 更新 SrmHandler 中的转换调用，使用新的 ErpToKingdeeConverter
- 优化了代码结构，提高了可维护性和扩展性

fishSteak committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

fishSteak committed:
build:调整 Redis配置并修改定时任务设置- 移除 Redis 配置中的 task.scheduling.simple.concurrency-limit 设置
- 在本地和测试环境配置中禁用定时任务调度- 在生产环境配置中启用定时任务调度

fishSteak committed:
feat(srm): 采购订单执行状态动态调整

- 新增执行状态调整功能，根据到货数量动态变更订单项执行状态
- 修改订单项执行状态机，增加到货数量调整事件
-优化订单项入库状态初始化逻辑- 更新执行状态枚举描述，更准确反映状态含义

fishSteak committed:
feat(srm): 采购入库单部分入库状态更新

- 新增实际入库数量字段并更新相关 VO 和 DO
- 优化采购入库单部分入库状态的更新逻辑- 调整库存和订单项的入库状态

fishSteak committed:
refactor(srm): 更新采购入库相关功能

- 新增 DbConfig 配置类
- 修改 SrmPurchaseInService 接口名称
- 更新 SrmPurchaseInServiceImpl 实现类
- 调整 OrderItemInActionImpl 中的 mapper 使用
- 同步订单项执行状态
- 修改状态机相关代码

JisenCai committed:
refactor(wms): 数据结构重构

JisenCai committed:
fix(wms): wip提供tms可售量查询接口

fishSteak committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

JisenCai committed:
refactor(wms): 数据结构重构

fishSteak committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

fishSteak committed:
feat(srm): 供应商名称去重校验

- 在创建和更新供应商时增加名称去重校验
- 新增供应商名称重复错误码和异常处理
-优化供应商数据同步流程
-调整日志级别和格式

JisenCai committed:
feat(wms): tms查询可售数量接口

JisenCai committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

JisenCai committed:
fix(wms): 盘点权限字段调整

JisenCai committed:
fix(wms): 盘点权限字段调整

JisenCai committed:
refactor(wms): 逻辑库存数据重构

fishSteak committed:
refactor(srm): 重构采购模块字段名称

- 将 inId 修改为 arriveId，inTime 修改为 arriveTime，reviewComment 修改为 reviewAdvice
- 更新了相关 mapper、service、controller 和 VO 类中的字段引用
- 此修改统一了采购到货单相关字段的命名，提高了代码的一致性和可读性

fishSteak committed:
refactor(tms): 重构部分代码并统一审核意见字段

- 将 reviewComment 字段统一更改为 auditAdvice，包括相关 DTO、VO 和数据库实体类
- 更新相关方法和接口中的字段引用- 优化部分代码结构，提高可读性和可维护性

fishSteak committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

fishSteak committed:
refactor(srm): 移除采购订单中的采购状态字段

- 从 SrmPurchaseOrderDO 类中删除 status 字段
-移除 SrmSupplierSaveReqVO 类中多个字段的 example 属性

JisenCai committed:
refactor(wms): 待上架数量字段重构

JisenCai committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

fishSteak committed:
refactor(srm): 重构采购模块审核逻辑

- 将 reviewAdvice 字段统一改为 auditAdvice，以提升代码一致性
- 优化了多个类中的审核意见设置逻辑
- 调整了部分变量名称以更好地反映其含义- 移除了冗余代码并优化了代码结构

fishSteak committed:
refactor(srm): 优化库存调整相关代码

- 修改 SrmEventEnum 中的 STOCK_ADJUSTMENT 描述为 "入库数量调整"- 修正 SrmPurchaseReturnApiImpl 中的事件触发为 OUT_STORAGE_ADJUSTMENT
- 优化 SrmPurchaseReturnedItemStatusMachine 中的状态机配置

JisenCai committed:
fix(wms): 出库单审批流逻辑调整

fishSteak committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

fishSteak committed:
refactor(srm): 重构采购模块相关代码

- 修改字段名称：将 actTaxPrice 改为 grossPrice，inStatus 改为 inboundStatus，xcode 改为 fbaCode
- 更新相关方法和接口中的字段引用
-调整部分逻辑以适应新的字段名称

JisenCai committed:
fix(wms): 文件名调整

fishSteak committed:
refactor(srm): 将 inStatus 字段重命名为 inboundStatus- 在多个实体类和 mapper 中，将 inStatus 字段重命名为 inboundStatus
- 更新了相关的 VO 和 DTO 中的字段名称
- 修改了状态机事件触发时使用的字段

fishSteak committed:
refactor(srm): 优化采购入库流程和状态机逻辑

- 暂停订单项状态机执行，因到货数量和入库数量分离
- 修复采购退货 WMS 出库单创建失败时的异常处理
- 优化采购入库单状态机事件传递逻辑
- 移除 WMS出库单保存请求中的冗余字段
- 改进 WMS 出库单生成逻辑，优化库存查询方式

fishSteak committed:
refactor(ERP-SRM-WMS): 重构相关模块代码

- 优化了 ERP 模块中的产品 API 实现
- 改进了 SRM 模块中的采购退货相关代码
- 调整了 WMS 模块中的出库单生成逻辑
-修复了各模块中的一些潜在问题

fishSteak committed:
refactor(srm): 更新出库数量调整事件名称

- 将事件名称从 RETURN_ADJUSTMENT 修改为 OUT_STORAGE_ADJUSTMENT
- 优化代码可读性和准确性

JisenCai committed:
fix(wms): 出库逻辑

JisenCai committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

fishSteak committed:
refactor(srm): 重构采购退货功能

- 移除订单项库存状态机联动逻辑
- 优化采购订单退货数量更新逻辑
- 调整采购退货单审核和退款流程
- 修改状态机名称和事件处理逻辑- 优化日志输出和异常提示信息

JisenCai committed:
fix(wms): 出库逻辑调整

JisenCai committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

JisenCai committed:
fix(wms): 出库逻辑调整

JisenCai committed:
fix(wms): 日志导出功能修复&代码清理

fishSteak committed:
fix(srm): 修复采购入库单支付状态相关问题

- 在 InPayActionImpl 中增加对子项支付状态为空的处理，避免出现空指针异常
-调整 SrmPurchaseInServiceImpl 中的订单项校验逻辑，提高代码效率

fishSteak committed:
refactor(somle): 重构采购入库单请求对象初始化方式- 将 KingdeePurInboundReqVO 的初始化方式从使用 Lombok 的 Builder 改为直接使用 new 关键字
-移除了 KingdeePurInboundReqVO 类中的 @Builder 注解

JisenCai committed:
fix(wms): 外部接口新增部门和公司非空校验

fishSteak committed:
refactor(srm): 优化采购退货业务逻辑

- 在生成出库单时，将主单的公司 ID 复制给明细行
- 在生成出库单明细时，添加归属部门 ID 和公司 ID

fishSteak committed:
refactor(srm): 优化退货单创建逻辑

- 移除了 SrmPurchaseReturnSaveReqVO 中的冗余字段
- 在 SrmPurchaseReturnServiceImpl 中优化了退货单项的创建过程
-通过从入库项复制相关字段，简化了数据处理流程

JisenCai committed:
fix(wms): Excel调库存归属

fishSteak committed:
refactor(srm): 重构采购模块税率相关逻辑

- 将 taxPercent 改为 taxRate，taxPrice 改为 tax，allAmount 改为 grossTotalPrice- 更新了相关 VO、DTO、DO 类中的字段名称
-调整了税率计算逻辑- 优化了部分代码结构和命名

fishSteak committed:
feat(srm): 优化供应商信息展示及查询功能

- 新增供应商付款条款信息展示
-优化付款条款字段类型，从 String 改为 Long
- 新增 getPaymentTermList 接口，支持批量获取付款条款信息
- 修复订单查询接口中的付款条款字段错误
- 优化供应商列表和详情接口，增加付款条款信息

fishSteak committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

JisenCai committed:
refactor(wms): 良次品替换数据重构

JisenCai committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

fishSteak committed:
feat(purchase): 优化采购退货功能

- 新增退货数量校验，确保退货数量不超过入库数量
- 添加出库状态初始化逻辑
- 优化退货单创建和审核流程
- 调整付款条款相关字段类型

fishSteak committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

JisenCai committed:
refactor(wms): barCode替换成productCode代码重构

JisenCai committed:
fix(wms): 换货权限字符调整

JisenCai committed:
Revert "refactor(wms): barCode替换成productCode代码重构"

This reverts commit 2c413dfe2d2cdbbcc27edd0813bff9e06dbffaf5.

fishSteak committed:
refactor(srm): 采购订单查询支持多选入库状态

- 在 SrmPurchaseOrderItemMapper 和 SrmPurchaseOrderMapper 中，将 inboundStatus 的查询条件从 eqIfPresent 改为 inIfPresent- 在 SrmPurchaseOrderPageReqVO 中，将 inboundStatus 从单个整数改为整数列表 inboundStatusList
- 这些修改允许在查询时指定多个入库状态，提高了查询的灵活性和可用性

fishSteak committed:
refactor(srm): 优化采购模块代码结构和逻辑

-调整代码结构，优化导入顺序
- 使用增强 for 循环替换索引访问
- 添加空值检查，提高代码健壮性
- 优化 WmsOutbound 相关操作逻辑
- 删除未使用的变量和注释

fishSteak committed:
feat(tms): 优化头程单和调拨单相关功能- 重构头程单状态更新逻辑，支持出库和入库信息回填
- 新增头程单明细更新功能，支持出库和入库数量更新
- 优化调拨单出库和入库逻辑，提高数据一致性和错误处理能力
- 为调拨单和明细添加创建人和更新人姓名字段，增强数据可读性

fishSteak committed:
build(dependencies): 添加 local-esb 配置文件并优化 TmsCustomRuleApiImpl

- 在 pom.xml 中添加 local-esb 配置文件，用于加载特定依赖- 在 TmsCustomRuleApiImpl 中使用 @Lazy 注解延迟加载 TmsCustomRuleService
-移除了未使用的注释代码

fishSteak committed:
refactor(tms): 重构头程单和申请单相关逻辑- 移除头程单明细中的包装长宽高、毛重和体积字段
- 在创建和更新头程单时计算总数量、总重量和总体积
- 从 ERP 系统获取产品包装信息并设置到明细项中
- 优化头程申请单合并逻辑
- 删除不必要的代码和注释

fishSteak committed:
feat(tms): 计算货柜净重并移除冗余代码

- 在 TmsFirstMileRequestServiceImpl 和 TmsFirstMileServiceImpl 中添加货柜净重计算逻辑
- 移除 TmsFirstMileSaveReqVO 中的 netWeight 字段
-优化代码结构，提高可读性和可维护性

fishSteak committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

JisenCai committed:
fix(wms): 更新在制数量接口

JisenCai committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

JisenCai committed:
fix(wms): 更新在制数量接口

fishSteak committed:
feat(srm): 采购订单审核时更新 WMS 仓库在制数量

- 在采购订单审核通过时，增加在制数量到 WMS 仓库
- 在采购订单反审核时，减少 WMS仓库的在制数量
- 新增 updateWareHouseGNumber 方法用于更新 WMS 仓库在制数量
- 在 reviewPurchaseOrder 方法中调用 updateWareHouseGNumber 方法

JisenCai committed:
refactor(wms): barCode调整为code(WMS模块范围)

fishSteak committed:
refactor(erp): 将 ErpProduct 中的 barCode 字段重命名为 code

- 更新了多个模块中与 ErpProduct 相关的代码，将 barCode替换为 code
- 修改了数据库表结构、API 接口和前端界面中对 barCode 的引用
- 重构了部分代码以适应新的字段名称

fishSteak committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

fishSteak committed:
refactor(srm): 将条码字段从 barCode 改为 productCode

- 修改了多个模块中的类和方法，将 barCode 字段统一改为 productCode
- 更新了相关的 API 接口和数据库表结构
- 调整了前端请求和响应参数- 修复了部分因字段变更导致的代码问题

fishSteak committed:
refactor(product): 重构产品代码相关逻辑

- 将 productDTO.getCode() 重命名为 productCode，提高代码可读性
- 更新相关方法中对产品代码的引用，使用 productCode 替代 barCode
- 修正部分逻辑判断，确保正确处理产品代码

fishSteak committed:
feat(tms): 为头程单明细添加库存信息

- 在 TmsFirstMileItemRespVO 中添加 stock 字段，用于存储库存信息
- 在 TmsFirstMileService 接口中添加 assembleTmsFirstMileStockRespVO 方法
- 在 TmsFirstMileServiceImpl 中实现 assembleTmsFirstMileStockRespVO 方法
- 更新 TmsFirstMileStockRespVO，添加公司名称和入库公司名称字段
- 移除 TmsFirstMileStockRespVO 中的冗余字段

JisenCai committed:
feature(wms): 换货单 获取仓库列表列表

JisenCai committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

fishSteak committed:
feat(tms): 为头程单明细添加库存信息

- 在 TmsFirstMileItemRespVO 中添加 stock 字段，用于存储库存信息
- 在 TmsFirstMileService 接口中添加 assembleTmsFirstMileStockRespVO 方法
- 在 TmsFirstMileServiceImpl 中实现 assembleTmsFirstMileStockRespVO 方法
- 更新 TmsFirstMileStockRespVO，添加公司名称和入库公司名称字段
- 移除 TmsFirstMileStockRespVO 中的冗余字段

fishSteak committed:
refactor(tms): 重构头程单编号校验逻辑

- 提取编号校验逻辑到单独的方法 validateFirstMileCode
- 增加参数 excludeId以支持更新时的编号校验
- 优化编号格式校验和最大序号设置逻辑
- 修改异常消息，提高错误提示的准确性

fishSteak committed:
feat(tms): 添加费用和船舶跟踪相关数据的校验

- 在 TmsFeeSaveReqVO 中添加了 costType 和 currencyType 的最小值校验
- 在 TmsVesselTrackingSaveReqVO 中添加了 toPort 和 fromPort 的非空和最小值校验- 这些校验增强了数据的完整性和可靠性

fishSteak committed:
refactor(srm): 优化采购申请单分页接口参数

- 在 SrmPurchaseRequestController 中，将 getPurchaseRequestPage 方法的请求体参数设置为非必须
- 在 TmsTransferApiImpl 中，添加日志记录调拨单出库和入库审核通过的信息
- 在 TmsTransferDO 中，更新出库状态和入库状态字段的注释，明确为 WMS 的状态字典

fishSteak committed:
refactor(crm): 修正国家代码字典类型

- 将 CrmDictTypeConstants.COUNTRY_CODE 修改为 CrmDictTypeConstants.CRM_COUNTRY_CODE
- 更新 CrmClueServiceImpl 中的字典类型引用
- 调整 application-local.yaml 中的日志级别配置- 移除 JsonUtilsX 中未使用的 parseObject 方法
-禁用 Tms 模块的几个测试类

fishSteak committed:
Merge branch 'refs/heads/somle-test' into somle-test-dev-TMS

# Conflicts:
#	somle-esb/src/main/java/com/somle/esb/converter/oms/ShopifyToOmsConverter.java

fishSteak committed:
refactor(erp): 修改产品代码查询相关方法

- 将 listProductIdByBarCode 方法重命名为 listProductIdByProductCode
- 更新方法参数名称从 barCode 到 productCode
- 修改相关控制器和 VO 类中的属性名称从 productBarCode 到 productCode
- 删除 SrmPurchaseRequestItemRespVO 中的产品条码字段

JisenCai committed:
fix(wms): 头程信息封装接口修复

JisenCai committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

c-tao committed:
fix: 删除erp模块与产品功能无关代码,

fishSteak committed:
refactor(tms): 优化头程单相关功能和错误提示- 完善头程单不存在错误提示，增加单号信息- 修正头程申请单ID描述
- 统一使用 @Validated 注解
- 优化部分代码结构

fishSteak committed:
Merge remote-tracking branch 'origin/somle-test-dev-TMS' into somle-test-dev-TMS

fishSteak committed:
refactor(fms): 重构 fms 模块的审核状态相关代码

- 新增 FmsAuditStatus 枚举类，用于统一管理 fms 模块的审核状态
- 修改 FmsFinancePaymentDO 和 FmsFinanceReceiptDO 中的审核状态字段类型
- 更新 FmsFinancePaymentServiceImpl 中使用新的审核状态枚举
- 删除 ErpProductService 中未使用的 listProductIdByProductCode 方法

c-tao committed:
fix: 删除ErpAuditStatus
